solidity/contracts/converter/ConverterBase.sol

```diff
@@ -474,6 +532,29 @@ contract ConverterBase is IConverter, TokenHandler, TokenHolder, ContractRegistr
 		return _targetAmount.mul(conversionFee).div(CONVERSION_FEE_RESOLUTION);
 	}
 
+	/**
+	 * @dev returns the protocol fee for a given target amount
+	 *
+	 * @param _targetAmount target amount
+	 *
+	 * @return calculated protocol fee
+	 */
+	function calculateProtocolFee(address _targetToken, uint256 _targetAmount) internal view returns (uint256) {
+		uint256 _protocolFee = getProtocolFeeFromSwapNetwork();
+		uint256 calculatedProtocolFee = _targetAmount.mul(_protocolFee).div(1e20);
+		protocolFeeTokensHeld[_targetToken] = protocolFeeTokensHeld[_targetToken].add(calculatedProtocolFee);
```

Comment:
function is marked as `view` but it is not.

Reply:
For compiler 0.4.26
If a function visibility declared as view, but in case there is a storage modification inside, I think it will keep modifying the storage.
CMIIW.
Anw, I will remove the _view_ keyword

Reply:
yeah, that was the behavior of `view` initially.